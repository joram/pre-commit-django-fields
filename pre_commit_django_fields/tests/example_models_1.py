from django.db import models


class ExampleModel(models.Model):
    id = models.UUIDField(primary_key=True)
    name = models.CharField(max_length=255)
    description = models.TextField()
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    def hello(self):
        match "var":
            case "var":
                print("Hello World!")
