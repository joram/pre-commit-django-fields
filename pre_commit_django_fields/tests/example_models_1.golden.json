[
  {
    "name": "id",
    "field_type": "UUIDField",
    "class_name": "ExampleModel",
    "filename": "example_models_1.py",
    "expected_field_classes": null,
    "lineno": 5
  },
  {
    "name": "name",
    "field_type": "CharField",
    "class_name": "ExampleModel",
    "filename": "example_models_1.py",
    "expected_field_classes": null,
    "lineno": 6
  },
  {
    "name": "description",
    "field_type": "TextField",
    "class_name": "ExampleModel",
    "filename": "example_models_1.py",
    "expected_field_classes": null,
    "lineno": 7
  },
  {
    "name": "created_at",
    "field_type": "DateTimeField",
    "class_name": "ExampleModel",
    "filename": "example_models_1.py",
    "expected_field_classes": null,
    "lineno": 8
  },
  {
    "name": "updated_at",
    "field_type": "DateTimeField",
    "class_name": "ExampleModel",
    "filename": "example_models_1.py",
    "expected_field_classes": null,
    "lineno": 9
  }
]
